portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -68,15 +69,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     // Add each comment into the commentsList.
     for (Entity entity : results.asIterable()) {
       long id = entity.getKey().getId();
-      String commentText = (String) entity.getProperty(COMMENT_PARAM);
+      String commentText = (String) entity.getProperty(COMMENT_TEXT_PARAM);
       long timestamp = (long) entity.getProperty("timestamp");
 
       Comment comment = new Comment(id, commentText, timestamp);
       commentsList.add(comment);
     }
 
-    // Convert the commentsList to JSON.
-    String json = convertToJson(commentsList);
+    // Limit the number of comments.
+    // Will be hard-coded to 5 until the frontend actually 
+    // sends the number of comments via the GET function in comments.js.
+    int commentLimit = 5; // = getNumberComments(request);
+
+    // An invalid input will result in an error, so we let the user know.
+    if (commentLimit == -1) {
+      response.setContentType("text/html");
+      response.getWriter().println("Please enter an integer between 1 and 20.");
```

Comment:
Set an error code on the response. 4XX errors are for client errors. Use the docs below to select the appropriate response code.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status